PrismaDB-QueryAST/DML/Expressions/ScalarFunction.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace PrismaDB.QueryAST.DML
+{
+    public class ScalarFunction : Expression
+    {
+        public string FunctionName;
+        public List<Expression> Parameters;
+
+        public ScalarFunction(string functionName)
+        {
+            setValue(functionName, new List<Expression>());
+        }
+
+        public ScalarFunction(string functionName, List<Expression> parameters)
+        {
+            setValue(functionName, parameters);
+        }
+
+        public override Expression Clone()
+        {
+            var clone = new ScalarFunction(FunctionName, new List<Expression>(Parameters));
+
+            return clone;
+        }
+
+        public override bool Equals(object other)
+        {
```

Comment:
Can you format this function differently? IMO, _very_ hard to read as it is. Maybe, split it into steps. e.g.
```csharp
if (Parameters.Count != otherF.Parameters.Count)
    return false;
if(!FunctionName.Equals(otherF.FunctionName))
    return false;
...
```